drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -39,9 +39,12 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 #include <net/ieee802154_radio.h>
 
-#if defined(CONFIG_SOC_NRF5340_CPUAPP) && \
-	defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
+#if defined(CONFIG_SOC_NRF5340_CPUAPP) && defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
+#if defined(CONFIG_SPM_SERVICE_READ)
```

Comment:
Would it make sense to check `CONFIG_SPM` instead? As that is the equivalent of `CONFIG_BUILD_WITH_TFM`

Reply:
On the other hand, `spm_request_read` is only available when `CONFIG_SPM_SERVICE_READ` so I thought it would be better to depend on the setting which directly controls the functionality. I can change, however, if it's against our rules.

Reply:
Makes sense, no strong opinion :)